server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultExportTask.java

```diff
@@ -22,17 +22,105 @@
 import java.io.IOException;
 import java.util.Optional;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+import javax.mail.internet.AddressException;
 
 import org.apache.james.core.MailAddress;
 import org.apache.james.core.User;
+import org.apache.james.json.DTOModule;
+import org.apache.james.server.task.json.dto.TaskDTO;
+import org.apache.james.server.task.json.dto.TaskDTOModule;
 import org.apache.james.task.Task;
 import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.vault.dto.query.QueryDTO;
+import org.apache.james.vault.dto.query.QueryTranslator;
 import org.apache.james.vault.search.Query;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+
 class DeletedMessagesVaultExportTask implements Task {
 
+    static final String TYPE = "deletedMessages/export";
+
+    public static final Function<DeletedMessagesVaultExportTask.Factory, TaskDTOModule> MODULE = (factory) ->
+        DTOModule
+            .forDomainObject(DeletedMessagesVaultExportTask.class)
+            .convertToDTO(DeletedMessagesVaultExportTask.DeletedMessagesVaultExportTaskDTO.class)
+            .toDomainObjectConverter(dto -> {
+                try {
+                    return factory.create(dto);
```

Comment:
I might be wrong but we cannot just use here Throwing and sneakyThrow instead of a try catch to throw a RuntimeException?

Reply:
if i try this the compiler failed to infer the Throwing function to the functional interface DomainObjectConverter